.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,56 @@
+name: gh-pages
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  skip:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "Skip job"
+
+  before:
+    runs-on: ubuntu-latest
+    if: "! contains(github.event.head_commit.message, '[skip ci]')"
+    steps:
+      - run: echo "not contains '[skip ci]'"
+
+  docs:
+    runs-on: ubuntu-latest
+    needs: before
+    env:
+      nim_version: '1.2.0'
+    steps:
+      - uses: actions/checkout@v1
+      - name: Cache choosenim
+        id: cache-choosenim
+        uses: actions/cache@v1
+        with:
+          path: ~/.choosenim
+          key: ${{ runner.os }}-choosenim-${{ env.nim_version }}
+      - name: Cache nimble
+        id: cache-nimble
+        uses: actions/cache@v1
+        with:
+          path: ~/.nimble
+          key: ${{ runner.os }}-nimble-${{ hashFiles('*.nimble') }}
+      - uses: jiro4989/setup-nim-action@v1.0.2
+        with:
+          nim-version: ${{ env.nim_version }}
+
+      - name: Generate API documents
+        run: |
+          for m in cligen cligen/parseopt3 cligen/argcvt; do
```

Comment:
Instead, just do `nim doc --project --outdir:docs cligen.nim`?

Reply:
It occurs error. And I could not resolve this errors.

```bash
⟩ nim doc --index:on --project --out:out --hints:off cligen.nim
/home/vagrant/src/github.com/jiro4989/cligen/cligen/sysUt.nim(20, 28) Error: '`' expected
/home/vagrant/src/github.com/jiro4989/cligen/cligen.nim(8, 45) Error: cannot export: postInc
/home/vagrant/src/github.com/jiro4989/cligen/cligen.nim(8, 54) Error: cannot export: delItem
/home/vagrant/src/github.com/jiro4989/cligen/cligen.nim(8, 63) Error: cannot export: fromNimble
/home/vagrant/src/github.com/jiro4989/cligen/cligen.nim(9, 8) Error: cannot export: summaryOfModule
/home/vagrant/src/github.com/jiro4989/cligen/cligen.nim(9, 25) Error: cannot export: docFromModuleOf
/home/vagrant/src/github.com/jiro4989/cligen/cligen.nim(9, 42) Error: cannot export: docFromProc
/home/vagrant/src/github.com/jiro4989/cligen/cligen/clCfgInit.nim(41, 11) Error: undeclared identifier: 'textAttrAlias'
/home/vagrant/src/github.com/jiro4989/cligen/cligen/clCfgInit.nim(41, 24) Error: attempting to call routine: 'textAttrAlias'
  found 'textAttrAlias' of kind 'unknown'
/home/vagrant/src/github.com/jiro4989/cligen/cligen/clCfgInit.nim(41, 24) Error: attempting to call routine: 'textAttrAlias'
  found 'textAttrAlias' of kind 'unknown'
/home/vagrant/src/github.com/jiro4989/cligen/cligen/clCfgInit.nim(41, 24) Error: expression 'textAttrAlias' cannot be called
/home/vagrant/src/github.com/jiro4989/cligen/cligen/clCfgInit.nim(59, 46) template/generic instantiation of `in` from here
/home/vagrant/src/github.com/jiro4989/cligen/cligen/clCfgInit.nim(59, 46) template/generic instantiation of `in` from here
/home/vagrant/.choosenim/toolchains/nim-#devel/lib/pure/collections/critbits.nim(25, 14) Error: internal error: ReplaceTypeVarsN: when condition not a bool
No stack traceback available
To create a stacktrace, rerun compilation with './koch temp doc <file>', see https://nim-lang.github.io/Nim/intern.html#debugging-the-compiler for details

```

Reply:
May be the dependencies of cligen cannot be found in the path.. I'll try out this command when I am at a computer. May be it needs just setting the paths in config.nims or something.

Reply:
Well, `nim doc cligen.nim` works fine.  It is true that the `sysUt.nim:20` error is probably malformed rST.  I wasn't sure how to quote a stropped operator.

The `clCfginit.nim` errors are probably because it is an `include` file and we may need to have some kind of `exclude this module from auto-doc generation` setting.  (If there isn't one then that is another issue to raise, IMO.)

The "internal" error from `critbits`...Well, that is the downside of auto-doc-updating..that you are more vulnerable to whatever bugs/limitations of the day.

Reply:
I changed reST docs.
https://github.com/c-blake/cligen/pull/153/commits/d2e96ba9d492a9e74c43b94388555d4a4a386feb

And success generation.

```bash
⟩ nim doc --project --outdir:docs cligen.nim
Hint: used config file '/home/vagrant/.choosenim/toolchains/nim-#devel/config/nim.cfg' [Conf]
Hint: used config file '/home/vagrant/.choosenim/toolchains/nim-#devel/config/config.nims' [Conf]
Hint: used config file '/home/vagrant/.choosenim/toolchains/nim-#devel/config/nimdoc.cfg' [Conf]
.....................................
Hint: 87076 lines; 0.506s; 94.891MiB peakmem; Debug build; proj: /home/vagrant/src/github.com/jiro4989/cligen/cligen.nim; out: /home/vagrant/src/github.com/jiro4989/cligen/docs/cligen.html [SuccessX]
```

Reply:
Ok.  Yeah..I had been hoping to have in the \`||\` in typewriter font with the backquotes, but it doesn't matter so much.